CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyWorker.cc

```diff
@@ -445,11 +449,31 @@ void SiStripHitEfficiencyWorker::analyze(const edm::Event& e, const edm::EventSe
   edm::Handle<edmNew::DetSetVector<SiStripCluster>> theClusters;
   e.getByToken(clusters_token_, theClusters);
 
-  edm::Handle<DetIdCollection> fedErrorIds;
-  e.getByToken(digis_token_, fedErrorIds);
+  // get the list of module IDs with FED-detected errors
+  //  - In Aug-2023, the data format was changed from DetIdCollection to DetIdVector.
+  //  - To provide some level of backward-compatibility,
+  //    the plugin checks for both types giving preference to the new format.
+  //  - If only the old format is available, the collection is
+  //    converted to the new format, then used downstream.
+  auto const& fedErrorIdsCol_h = e.getHandle(digisCol_token_);
+  auto const& fedErrorIdsVec_h = e.getHandle(digisVec_token_);
+  DetIdVector fedErrorIdsVecFromColl;
+  if (not fedErrorIdsVec_h.isValid()) {
```

Comment:
out of curiosity I have tried myself a similar approach but using `failedToGet` instead of `isValid`, what's the actual difference between the two? maybe some core expert could educate me on that (apologies I could not find in the manual an answer).

Reply:
According to a comment in `DataFormats/Common/interface/Handle.h`:

> To check validity, one can use the `isValid()` function.
> 
> If `failedToGet()` returns true then the requested data is not available
> If `failedToGet()` returns false but `isValid()` is also false then no attempt 
>   to get data has occurred

Reply:
(assuming it is still accurate)